discoverer/src/main/java/eu/seaclouds/platform/discoverer/core/LocationMapping.java

```diff
@@ -18,28 +18,34 @@
 
 package eu.seaclouds.platform.discoverer.core;
 
+import java.io.*;
 import java.util.HashMap;
 
 public class LocationMapping {
 
-    private static HashMap<String, String> map;
+    private static HashMap<String, String> map = new HashMap<>();
 
-    static {
-        HashMap<String, String> initializedMap = new HashMap<>();
+    public static void initializeMap(InputStream resource) {
+        try {
+            InputStreamReader isr = new InputStreamReader(resource);
+            BufferedReader br = new BufferedReader(isr);
+            String line;
 
-        initializedMap.put("Amazon_EC2", "aws-ec2");
-        initializedMap.put("SoftLayer_Cloud_Servers", "softlayer");
-        initializedMap.put("Microsoft_Azure_Virtual_Machines", "azurecompute");
-        initializedMap.put("Google_Compute_Engine", "google-compute-engine");
-        initializedMap.put("HP_Cloud_Compute", "hpcloud-compute");
-        initializedMap.put("Cloud_Foundry", "CloudFoundry");
-
-        //initializedMap.put("Rackspace_Cloud_Servers", "");
-        // Rackspace provides more location so it is not yes possible to make an unique mapping
+            while ((line = br.readLine()) != null) {
+                String parts[] = line.split("=");
+                map.put(parts[0], parts[1]);
+            }
 
-        map = initializedMap;
+            br.close();
```

Comment:
As we are using Java7, maybe this could be helpful https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html